packages/nvidia_nat_a2a/tests/server/test_front_end_config_security.py

```
@@ -0,0 +1,63 @@
+
+    @pytest.mark.parametrize("host", ["localhost", "127.0.0.1", "::1"])
+    def test_localhost_bind_without_auth_is_allowed(self, host):
+        """Localhost binds don't need server_auth — loopback is not network-accessible."""
+        cfg = A2AFrontEndConfig(host=host)
+        assert cfg.host == host
+        assert cfg.server_auth is None
+
+    def test_non_localhost_bind_without_auth_is_rejected(self):
+        """0.0.0.0 (or any non-loopback) without server_auth must fail validation."""
+        with pytest.raises(ValidationError) as exc_info:
+            A2AFrontEndConfig(host="0.0.0.0")
```

Ruff (0.15.10)

[error] Possible binding to all interfaces

(S104)